lib/services/local/databaseobject_test.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package local
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	dbobjectv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobject/v1"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	"github.com/gravitational/teleport/lib/srv/db/common/databaseobject"
+)
+
+// TestDatabaseObjectCRUD tests backend operations with DatabaseObject resources.
+func TestDatabaseObjectCRUD(t *testing.T) {
```

Comment:
Nit: I find these super long "CRUD" tests which check all the methods super hard to read and modify in future. I'd definitely prefer a test per method or at least some use of subtests to break this up - I'm aware that most of the tests in this package are in this style but this feels legacy to me.

Reply:
Done: 06f775d3fc0356fc69784429299d4ccc0268e7dc

Reply:
Way more readable! Thank you!